sbi/inference/snle/snle_base.py

```diff
@@ -208,8 +209,10 @@ def train(
                 )
                 # Evaluate on x with theta as context.
                 log_prob = self._neural_net.log_prob(x_batch, context=theta_batch)
-                loss = -torch.mean(log_prob)
-                loss.backward()
+                batch_loss = -torch.mean(log_prob)
+                batch_log_probs_sum -= batch_loss.item()
```

Comment:
to avoid confusion due to the double minus, how about
```
batch_log_probs = torch.mean(log_prob)
batch_log_probs_sum += batch_log_probs
batch_loss = -batch_log_probs
```
?

Reply:
or add a comment for explanation.

Reply:
I agree that the double minus is confusing. But it has a reason: I wanted to keep the variable names consistent between all three methods (SNPE, SNLE, SNRE). Both SNRE and SNPE use the `._loss()` to compute the `batch_loss`.

Reply:
I'll add a comment though